incalmo/c2server/c2server.py

```
@@ -244,6 +245,60 @@ def agent_download():
         return jsonify({"error": f"Internal server error: {str(e)}"}), 500
 
 
+# Stream logs
+@app.route("/stream_logs", methods=["GET"])
+def stream_logs():
+    def generate_log_stream():
+        # Send headers for SSE
+        yield "retry: 1000\n\n"
+
+        # Get the latest output directory
+        output_dirs = sorted(Path("output").glob("*_*-*-*"), reverse=True)
+        if not output_dirs:
+            yield f"data: {json.dumps({'error': 'No log directories found'})}\n\n"
+            return
+
+        latest_dir = output_dirs[0]
+        actions_log_path = latest_dir / "actions.json"
+
+        if not actions_log_path.exists():
+            yield f"data: {json.dumps({'error': 'Actions log file not found'})}\n\n"
+            return
+
+        position = 0
+
+        while True:
+            try:
+                with open(actions_log_path, "r") as f:
+                    f.seek(position)
+                    # Read any new lines
+                    for line in f:
+                        try:
+                            log_entry = json.loads(line)
+                            if log_entry.get("type") == "LowLevelAction":
+                                yield f"data: {line.strip()}\n\n"
+                        except json.JSONDecodeError:
+                            continue
+                    # Update position for next read
+                    position = f.tell()
+
+            except Exception as e:
```

Avoid catch all exceptions